lib/pushmi_pullyu.rb

```diff
@@ -1,12 +1,63 @@
 require 'pushmi_pullyu/logging'
 
 require 'pushmi_pullyu/cli'
-require 'pushmi_pullyu/config'
 require 'pushmi_pullyu/preservation_queue'
 require 'pushmi_pullyu/version'
 require 'active_support'
 require 'active_support/core_ext'
 
 # PushmiPullyu main module
 module PushmiPullyu
+  LOGFILE = 'log/pushmi_pullyu.log'.freeze
+  PIDDIR  = 'tmp/pids'.freeze
+  PROCESS_NAME = 'pushmi_pullyu'.freeze
+
+  DEFAULTS = {
+    daemonize: false,
+    debug: false,
+    logfile: LOGFILE,
+    minimum_age: 0,
+    monitor: false,
+    piddir: PIDDIR,
+    process_name: PROCESS_NAME,
+    redis: {
+      host: 'localhost',
+      port: 6379,
+      queue_name: 'dev:pmpy_queue'
+    },
+    # TODO: rest of these are examples for solr/fedora/swift... feel free to fill them in correctly
+    solr: {
+      url: 'http://localhost:8983/solr/hydra-development'
+    },
+    fedora: {
+      url: 'http://localhost:8984/rest',
+      user: 'fedoraAdmin',
+      password: 'fedoraAdmin',
+      base_path: '/dev'
+    },
+    swift: {
+      auth_version: 2.0,
+      tenant: 'Millennium Falcon',
+      username: 'han',
+      password: 'YT-1300',
+      endpoint: 'https//corellia.lan',
+      temp_url_key: '492727ZED'
+    }
+  }.freeze
+
+  def self.options
+    @options ||= DEFAULTS.dup
+  end
+
+  def self.options=(opts)
+    @options = opts
```

Comment:
Should this do a `.reverse_merge` with defaults so that you can set only some options and let the rest inherit the defaults? It would let you not handle the merging out in the parse command, which makes this a little cleaner.